src/examples/rest/src/backend/schema/auth/forgotten-password/resolver.ts

```diff
@@ -0,0 +1,53 @@
+import {
+	createForgottenPasswordAuthResolver,
+	Credential,
+	ForgottenPasswordLinkData,
+	UserProfile,
+} from '@exogee/graphweaver-auth';
+import { Filter, graphweaverMetadata, Resolver } from '@exogee/graphweaver';
+import { MikroBackendProvider } from '@exogee/graphweaver-mikroorm';
+
+import { myConnection } from '../../../database';
+import {
+	Credential as OrmCredential,
+	Authentication as OrmAuthentication,
+} from '../../../entities/mysql';
+import { ForgottenPasswordLink } from './entity';
+
+@Resolver()
+export class ForgottenPasswordLinkResolver extends createForgottenPasswordAuthResolver<
+	OrmAuthentication<ForgottenPasswordLinkData>
+>(
+	ForgottenPasswordLink,
+	new MikroBackendProvider(OrmAuthentication<ForgottenPasswordLinkData>, myConnection)
+) {
+	/**
+	 * A callback that can be used to send the forgotten link via channels such as email or SMS
+	 * @param url the URL that was generated and should be sent to the user
+	 * @param forgotPasswordLink the forgotten password link entity that was generated
+	 * @returns a boolean to indicate that the URL has been sent
+	 */
+	async sendForgottenPasswordLink(url: URL): Promise<boolean> {
+		// In a production system this would email / sms the forgotten link and you would not log to the console!
+		console.log(`\n\n ######## ForgotPasswordLink: ${url.toString()} ######## \n\n`);
+		return true;
+	}
+
+	/**
+	 *
+	 * @param username fetch user details using a username
+	 * @returns return a UserProfile compatible entity
+	 */
+	async getUser(username: string): Promise<UserProfile> {
+		//@todo - extend the base loaders to handle filters - find by filter.
```

Comment:
Why is this needed? If it is needed, should we do it as part of this PR or remove the @todo? I'm pretty sure the base loaders can already find with a filter.

Reply:
No our Base loaders only take an ID and here we have a username. 

I think we can remove the todo in this case as we are no using the base loader but the provider directly.

Reply:
Yup, sounds good!